16.0 mig commown factor out contract payment issues

Honeyxilia opened the pull request.

Honeyxilia committed:
Create contract_payment_issues to show payment issues on a contract

Honeyxilia committed:
Moved out code from commown related to payment issues in contracts

Honeyxilia committed:
[test] Added a test for the computation method

Honeyxilia committed:
Assigned result of project.task without calling the ids attribute

From the web UI, trying to access the ids of the project.task search causes a cache issue ("Compute method failed to assign").
As such, we remove the ids attribute to allow a direct assignment, since the search method returns a recordset.

fcayre commented on contract_payment_issues/__manifest__.py:
```diff
@@ -12,6 +12,10 @@
     "depends": [
-        "base",
+        "payment_slimpay_issue",
+        "commown_contractual_issue",
```
Là encore, cette dépendance n'est pas logique et semble à notre portée : si je ne me trompe pas elle n'est utilisée que pour placer correctement le champ payment_issues à côté de issue_ids, c'est dommage.
Pour pallier ce problème je pense qu'il faut créer un 3ème module qui dépend des 2 précédents et déplace les champs (mis de base un peu n'importe où, par exemple chacun dans son propre onglet) pour rendre la vue actuelle.

Honeyxilia replied:
On peut en effet créer un module unificateur qui réunisse les différents issues dans une même page - cependant, si l'on souhaite les mettre les deux catégories dans leur onglet séparé, alors il suffirait juste de retirer la dépendance à `commown_contractual_issue`, mais cela changerait l'interface par rapport à la v12

Ainsi, je crée un module qui met les deux issues dans un même onglet sur un contrat donné

Honeyxilia replied:
J'ai ainsi fait en sorte que chaque module individuel crée sa propre page, puis via un 3e module, la vue déplace les champs afin de les unir en une seule page c:

fcayre reviewed (changes requested):
Voir le commentaire sur les dépendances dans __manifest__.py

Honeyxilia committed:
Created new module commown_contract_unique_issues_tab

Honeyxilia committed:
Removed commown_contractual_issue dependency in regards to field placement

Since this module doesn't depend on commown_contractual_issue for any other feature other than its placement,
we can easily place the payment_issues field in its own dedicated notebook page.

To reunite both issues fields in the same page, we will make a unifying module to move both fields in the same page.

Honeyxilia committed:
Added contract form view overload to unite both issues-related fields

fcayre commented on contract_payment_issues/models/contract.py:
```diff
@@ -12,8 +12,6 @@ class PaymentIssuesContract(models.Model):
-                .ids
+            record.payment_issue_ids = self.env["project.task"].search(
+                [("invoice_id", "in", record._get_related_invoices().ids)]
             )
```
Je suis surpris ici que l'utilisation d'une odoo.Command ne soit pas nécessaire.

Honeyxilia replied:
Où est-ce qu'il faudrait utiliser Command, étant donné qu'il ne s'agit que d'une recherche ?

fcayre commented on commown_contract_unique_issues_tab/README.rst:
```diff
@@ -0,0 +1,72 @@
+
+This module is part of the `commown/commown-odoo-addons <https://github.com/commown/commown-odoo-addons/tree/16.0/commown_contract_unique_issues_tab>`_ project on GitHub.
+
+You are welcome to contribute.
```
Je pense qu'on peut renommer ce type de module view_* (sur le modèle du test_* qu'on vient de créer). Ici : view_contract_issues.

Ça te semble bien ?

Honeyxilia replied:
ça me semble bien, et je l'ai appliqué c:

fcayre reviewed (changes requested):
J'ai une suggestion de nommage de ce type de modules, qui pourrait devenir assez courant si on veut rendre l'arbre de dépendances plus sain.

Honeyxilia commented:
/ocabot rebase

Honeyxilia commented:
/ocabot merge patch
